provisioning/miniprov/hare_mp/main.py

```diff
@@ -445,10 +445,36 @@ def start_mkfs_parallel(hostname: str, hare_config_dir: str):
                       f'{perf_result}\n\n')
 
 
+@repeat_if_fails()
+def wait_for_mkfs_to_complete_on_all_nodes(utils: Utils,
+                                           cns_utils: ConsulUtil):
+    nodes = utils.get_io_nodes()
+    for node in nodes:
+        while not cns_utils.kv.kv_get(f'mkfs_done/{node}', recurse=True):
+            pass
+
+
+@repeat_if_fails()
+def cleanup_mkfs_state(utils: Utils, cns_utils: ConsulUtil):
+    hostname = utils.get_local_hostname()
+    keys: List[KeyDelete] = [
+        KeyDelete(name=f'mkfs_done/{hostname}', recurse=True),
+    ]
+
+    if not cns_utils.kv.kv_delete_in_transaction(keys):
+        pass
```

Comment:
Don't we need to log anything? Is this scenario expected (I mean the situation when transaction failed)?

Reply:
Yeah, sleep is a good idea, let me do that. 
Subset of transactions can fail as mentioned at https://www.consul.io/api-docs/txn#sample-response.
But I am not sure if we check every transaction's response, but in general, yeah, it is a good idea to post atleast a log message in case of failure. Let me add that too.

Reply:
Also the function does return false in case of Client Error,
```
    def kv_delete_in_transaction(self, tx_payload: List[KeyDelete]) -> bool:
        def to_payload(v: KeyDelete) -> Dict[str, Any]:
            return {'KV': {'Key': v.name, 'Verb':
                           'delete-tree' if v.recurse else 'delete'}}

        try:
            self.cns.txn.put([to_payload(i) for i in tx_payload])
            return True
        except ClientError:
            # If a transaction fails, Consul returns HTTP 409 with the
            # JSON payload describing the reason why the transaction
            # was rejected.
            # The library transforms HTTP 409 into generic ClientException.
            # Unfortunately, we can't easily extract the payload from it.
            return False
        except (ConsulException, HTTPError, RequestException) as e:
            raise HAConsistencyException(f'Failed to delete key(s)'
                                         f' from KV, error: {e}')
```